-1

Tested at: b8e7ce95faae9d3ea4b8f5a6e565a9238bc2043f

CMSSW: CMSSW_11_2_X_2020-10-02-1100
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-05f749/9708/summary.html

I found follow errors while testing this PR

**Failed tests:  Build**

* **Build**:

I found compilation error when building: 

<pre>--- Registered EDM Plugin: GeometryVeryForwardGeometryBuilderPlugins
Entering library rule at Geometry/VeryForwardGeometryBuilder
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-10-02-1100/src/Geometry/VeryForwardGeometryBuilder/src/CTPPSGeometry.cc
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-10-02-1100/src/Geometry/VeryForwardGeometryBuilder/src/DetGeomDesc.cc
In file included from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-10-02-1100/src/Geometry/VeryForwardGeometryBuilder/src/DetGeomDesc.cc:17:
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-10-02-1100/poison/DetectorDescription/DDCMS/interface/DDShapes.h:1:2: error: #error THIS FILE HAS BEEN REMOVED FROM THE PACKAGE.
 #error THIS FILE HAS BEEN REMOVED FROM THE PACKAGE.
  ^~~~~
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-10-02-1100/src/Geometry/VeryForwardGeometryBuilder/src/DetGeomDescBuilder.cc
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-10-02-1100/src/Geometry/VeryForwardGeometryBuilder/src/module.cc
In file included from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-10-02-1100/src/Geometry/VeryForwardGeometryBuilder/src/DetGeomDesc.cc:17:
</pre>

--- a/MagneticField/GeomBuilder/src/DD4hep_volumeHandle.cc
+++ b/MagneticField/GeomBuilder/src/DD4hep_volumeHandle.cc
@@ -21,7 +21,6 @@ using namespace SurfaceOrientation;
 using namespace std;
 using namespace magneticfield;
 using namespace edm;
-//using namespace angle_units::operators;
 
 using DDBox = dd4hep::Box;
 using DDTrap = dd4hep::Trap;
@@ -78,7 +77,7 @@ volumeHandle::volumeHandle(const cms::DDFilteredView &fv, bool expand2Pi, bool d
       double rInPlusZ = cons.rMin2();
       double rOutPlusZ = cons.rMax2();
       double startPhi = cons.startPhi();
-      double deltaPhi = cons.endPhi() - startPhi;
+      double deltaPhi = reco::deltaPhi(cons.endPhi(), startPhi);
       buildCons(zhalf, rInMinusZ, rOutMinusZ, rInPlusZ, rOutPlusZ, startPhi, deltaPhi);
     } break;
     case DDSolidShape::ddtubs: {

